File: Assignment_1/lr_arr.c
#include <stdio.h>
int main()
{
    int n;
    printf("Enter size of array:");
    scanf("%d",&n);
    int arr[n], l_arr[n], r_arr[n];
    for(int i=0;i<n;i++)
    {
        printf("Enter element:");
        scanf("%d",&arr[i]);
    }
    int k;
    printf("Enter value of k:");
    scanf("%d",&k);
    while(k>0)
    {
        int temp=arr[0];
        for(int j=0;j<n-1;j++)
        {
            l_arr[j]=arr[j+1];
        }
        l_arr[n-1]=temp;
        k--;
    }
    printf("Left rotation array:");
    for(int i=0;i<n;i++)
    {
        printf("%d ",&l_arr[i]);
    }
    while(k>0)
    {
        int temp=arr[n];
        for(int j=0;j<n-1;j++)
        {
            r_arr[j+1]=arr[j];
        }
        r_arr[0]=temp;
        k--;
    }
    printf("Right rotation array:");
    for(int i=0;i<n;i++)
    {
        printf("%d ",arr[i]);
    }
    
    return 0;
}

File: Assignment_1/lr_arr.py
n=int(input("Enter size of array:"))
arr=[]
for i in range(n):
    arr.append(int(input("Enter element:")))
k=int(input("Enter value of k:"))
l_arr=arr[:]
r_arr=arr[:]
for _ in range(k):
    l_arr=l_arr[1:]+[l_arr[0]]
    k-=1
for _ in range(k):
    r_arr=[r_arr[-1]]+r_arr[:-1]
    cpy_k-=1
print("Left rotation array:",l_arr)
print("Right rotation array:",r_arr)